src/h3lib/lib/polygon.c

```diff
@@ -83,3 +83,103 @@ bool pointInsidePolygon(const GeoPolygon *geoPolygon, const BBox *bboxes,
+    if (cellBoundaryCrossesGeoLoop(&(geoPolygon->geoloop), &bboxes[0], boundary,
+                                   boundaryBBox)) {
+        return false;
+    }
+    // Check for line intersections with any hole
+    for (int i = 0; i < geoPolygon->numHoles; i++) {
+        if (cellBoundaryCrossesGeoLoop(&(geoPolygon->holes[i]), &bboxes[i + 1],
+                                       boundary, boundaryBBox)) {
+            return false;
+        }
+    }
+    return true;
+}
+
+/**
+ * Whether a cell boundary crosses a geo loop. Crossing in this case means
+ * whether any line segments intersect; it does not include containment.
+ * @param  geoloop  Geo loop to test
+ * @param  boundary Cell boundary to test
+ * @return          Whether any line segments in the boundary intersect any line
+ * segments in the geo loop
+ */
+bool cellBoundaryCrossesGeoLoop(const GeoLoop *geoloop, const BBox *loopBBox,
+                                const CellBoundary *boundary,
+                                const BBox *boundaryBBox) {
+    if (!bboxIntersects(loopBBox, boundaryBBox)) {
+        return false;
+    }
+    LongitudeNormalization loopNormalization;
+    LongitudeNormalization boundaryNormalization;
+    bboxNormalization(loopBBox, boundaryBBox, &loopNormalization,
+                      &boundaryNormalization);
+
+    CellBoundary normalBoundary = *boundary;
+    for (int i = 0; i < boundary->numVerts; i++) {
+        normalBoundary.verts[i].lng =
+            NORMALIZE_LNG(normalBoundary.verts[i].lng, boundaryNormalization);
+    }
+
+    LatLng loop1;
+    LatLng loop2;
+    for (int i = 0; i < geoloop->numVerts; i++) {
+        loop1 = geoloop->verts[i];
+        loop1.lng = NORMALIZE_LNG(loop1.lng, loopNormalization);
+        loop2 = geoloop->verts[(i + 1) % geoloop->numVerts];
+        loop2.lng = NORMALIZE_LNG(loop2.lng, loopNormalization);
+        for (int j = 0; j < normalBoundary.numVerts; j++) {
+            if (lineIntersectsLine(
+                    &loop1, &loop2, &normalBoundary.verts[j],
+                    &normalBoundary.verts[(j + 1) % normalBoundary.numVerts])) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+/**
+ * Whether two lines intersect. This is a purely Cartesian implementation
+ * and does not consider anti-meridian wrapping, poles, etc. Based on
+ * http://www.jeffreythompson.org/collision-detection/line-line.php
+ * @param  a1 Start of line A
+ * @param  a2 End of line A
+ * @param  b1 Start of line B
+ * @param  b2 End of line B
+ * @return    Whether the lines intersect
+ */
+bool lineIntersectsLine(const LatLng *a1, const LatLng *a2, const LatLng *b1,
+                        const LatLng *b2) {
+    double test;
+    test = ((b2->lat - b1->lat) * (a1->lng - b1->lng) -
+            (b2->lng - b1->lng) * (a1->lat - b1->lat)) /
+           ((b2->lng - b1->lng) * (a2->lat - a1->lat) -
+            (b2->lat - b1->lat) * (a2->lng - a1->lng));
```

Comment:
In CI, testPolygonInternal hits an unguarded division by zero on this line:
```
TEST polygonInternal
/Users/isaac/oss/h3/src/h3lib/lib/polygon.c:175:56: runtime error: division by zero
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /Users/isaac/oss/h3/src/h3lib/lib/polygon.c:175:56 in
......................................................................................zsh: abort      ./bin/testPolygonInternal
```

(to configure: `cmake -DCMAKE_C_FLAGS="-fsanitize=undefined,float-divide-by-zero -fno-sanitize-recover=undefined,float-divide-by-zero" ..`)

Reply:
Thanks! Will fix